pennylane/queuing.py

```diff
@@ -347,3 +346,34 @@ def circuit(x):
         context.append(op)
 
     return op
+
+
+# pylint: disable=protected-access
+def process_queue(queue):
+    """Process the annotated queue, creating a list of quantum
+    operations and measurement processes.
+
+    Args:
+        queue (.AnnotatedQueue): The queue to be processed into individual lists
+
+    Returns:
+        tuple[list(.Operation), list(.Operation), list(.MeasurementProcess)]:
+        The list of preparation operations, the list of main tape operations, and the list of tape measurements
+    """
+    lists = {"_prep": [], "_ops": [], "_measurements": []}
+    list_order = {"_prep": 0, "_ops": 1, "_measurements": 2}
+    current_list = "_prep"
+
+    for obj, info in queue.items():
+
+        if "owner" not in info and getattr(obj, "_queue_category", None) is not None:
+            if list_order[obj._queue_category] > list_order[current_list]:
+                current_list = obj._queue_category
+            elif list_order[obj._queue_category] < list_order[current_list]:
+                raise ValueError(
+                    f"{obj._queue_category[1:]} operation {obj} must occur prior "
+                    f"to {current_list[1:]}. Please place earlier in the queue."
+                )
+            lists[obj._queue_category].append(obj)
+
+    return lists["_prep"], lists["_ops"], lists["_measurements"]
```

Comment:
do we prefer this to be the same order as QuantumScript (aka move `_prep` to the end)?

Reply:
I think the ordering is different because people often don't set `_prep`, so you can just do `QuantumScript(ops, measurements)` and the prep will default to `[]`. If we change the order, we can do `QuantumScript(process_queue(q))` in one line - otherwise, we have to process the queue first, then initialize a script in the next line. The way it's currently implemented is of course the intuitive ordering of a circuit.

Reply:
So we do eventually 🤞 want to get rid of the distinction between normal ops and prep. Sticking it at the end will help with that.

Also, I like the idea of doing QuantumScript(*process_queue(q)).

Reply:
matching QuantumScript because it's cool and the future of _prep is precarious as it is